benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/scheduler/ExperimentTaskScheduler.java

```diff
@@ -155,10 +158,19 @@ public synchronized void handleRunningExperiment(String experimentID) {
           break;
         }
 
+        runningState = experimentModelDAO.getRunningState(experimentID);
+        logger.info("handleRunningExperiment: prevExperimentState == " + prevExperimentState);
+        logger.info("handleRunningExperiment: runningState == " + runningState);
+
         // Exit as soon as final state is executed
-        if (prevTestState != null && experimentState != null
-            && prevTestState.name().equals(experimentState.name())
-            && experimentState == TERMINATED) {
+        if (prevExperimentState == TERMINATED) {
```

Comment:
this doesn't work because we need to handle the `TERMINATED` state before exiting